apps/src/blockly/addons/contextMenu.js

```diff
@@ -286,22 +331,41 @@ export function registerHelp() {
     },
     scopeType: GoogleBlockly.ContextMenuRegistry.ScopeType.BLOCK,
     id: 'blockHelp',
-    weight: 11,
+    weight,
   };
   GoogleBlockly.ContextMenuRegistry.registry.register(helpOption);
 }
 
 const registerAllContextMenuItems = function () {
   unregisterDefaultOptions();
-  registerDeletable();
-  registerMovable();
-  registerEditable();
-  registerShadow();
-  registerUnshadow();
-  registerKeyboardNavigation();
-  registerDarkMode();
-  registerThemes(themes);
-  registerHelp();
+
+  // Expected registered block options (from Core and plugins):
+  // 0: blockCopyToStorage
+  // 2: blockComment
+  // 5: blockDisable
+  // 6: blockDelete
+  let nextBlockOptionWeight = 7;
+
+  // Expected registered workspace options (from Core and plugins):
+  // 0: blockPasteFromStorage
+  // 1: undoWorkspace
+  // 2: redoWorkspace
+  let nextWorkspaceOptionWeight = 3;
+
+  // Custom block options
+  registerHelp(nextBlockOptionWeight++);
+  registerDeletable(nextBlockOptionWeight++);
+  registerMovable(nextBlockOptionWeight++);
+  registerEditable(nextBlockOptionWeight++);
+  registerShadow(nextBlockOptionWeight++);
+  registerUnshadow(nextBlockOptionWeight);
+
+  // Custom workspace options
+  registerKeyboardNavigation(nextWorkspaceOptionWeight++);
+  registerAllCursors(nextWorkspaceOptionWeight, cursors);
+  nextWorkspaceOptionWeight += cursors.length;
+  registerDarkMode(nextWorkspaceOptionWeight++);
+  registerThemes(nextWorkspaceOptionWeight, themes);
```

Comment:
Take or leave an alternate organization idea here:

```
// TODO: Add comments on why magic numbers make sense
const INITIAL_BLOCK_OPTION_WEIGHT = 7;
const INITIAL_WORKSPACE_OPTION_WEIGHT = 3;

const registerAllContextMenuItems = function () {
  unregisterDefaultOptions();
  
  let nextBlockOptionWeight = INITIAL_BLOCK_OPTION_WEIGHT;
  registerCustomBlockOptions(nextBlockOptionWeight);

  let nextWorkspaceOptionWeight = INITIAL_WORKSPACE_OPTION_WEIGHT;
  registerCustomWorkspaceOptions(nextWorkspaceOptionWeight, cursors, themes);
};

function registerCustomBlockOptions(startWeight) {
  const blockOptions = [registerHelp, registerDeletable, registerMovable, registerEditable, registerShadow, registerUnshadow];
  blockOptions.forEach(registerOption => {
    registerOption(startWeight);
    startWeight += 1;
  });
}

function registerCustomWorkspaceOptions(startWeight, cursors, themes) {
  const workspaceOptions = [registerKeyboardNavigation, registerDarkMode];
  workspaceOptions.forEach(registerOption => {
    registerOption(startWeight);
    startWeight += 1;
  });

  registerAllCursors(startWeight, cursors);
  startWeight += cursors.length;

  // No explicit increment needed here since registerThemes internally handles the increment
  registerThemes(startWeight, themes);
}
```

Reply:
I'd probably avoid using pre/post-increment operators as an argument because it feels a little weird to me to remember the timing...
```
If used postfix, with operator after operand (for example, x++), the increment operator increments and returns the value before incrementing.
```
But that's just one person's opinion!